File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownChirpException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.R;
import com.github.dedis.popstellar.model.objects.security.MessageID;

public class UnknownChirpException extends GenericException {

  public UnknownChirpException(MessageID id) {
    super("Chirp with id " + id + " is unknown.");
  }

  @Override
  public int getUserMessage() {
    return R.string.unknown_chirp_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/UnknownChirpException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.R
import com.github.dedis.popstellar.model.objects.security.MessageID

class UnknownChirpException(id: MessageID) : GenericException("Chirp with id $id is unknown.") {
  override val userMessage: Int
    get() = R.string.unknown_chirp_exception

  override val userMessageArguments: Array<Any?>
    get() = arrayOfNulls(0)
}
